Yafc.Parser/Data/FactorioDataDeserializer_RecipeAndTechnology.cs

```diff
@@ -229,6 +229,29 @@ private Ingredient[] LoadResearchIngredientList(LuaTable table, string typeDotNa
         }).Where(x => x is not null).ToArray() ?? [];
     }
 
+    private void LoadResearchTrigger(LuaTable researchTriggerTable, ref Technology technology, ErrorCollector errorCollector) {
+        if (!researchTriggerTable.Get("type", out string? type)) {
+            errorCollector.Error($"Research trigger of {technology.typeDotName} does not have a type field", ErrorSeverity.MinorDataLoss);
+            return;
+        }
+
+        switch (type) {
+            case "craft-item":
+                if (!researchTriggerTable.Get("item", out string? craftItemName)) {
+                    errorCollector.Error($"Research trigger craft-item of {technology.typeDotName} does not have a item field", ErrorSeverity.MinorDataLoss);
+                    break;
+                }
+                researchTriggerTable.Get("count", out float craftCount, 1);
```

Comment:
minor code style - I don't know whether we even have some conventions for it, but I have seen too many out parameters lately

`float craftCount = researchTriggerTable.Get("count", 1f);` wouldn't need the out parameter, which is always a bit awkward and doesn't discard the bool result

Reply:
I just copied from another place as there are so/too many variants of the `LuaTable.Get()` function.

Reply:
Resolving this conversation to merge the PR. Please reopen if necessary.